### PR TITLE
AvroIO compression support

### PR DESCRIPTION
**Pull request by ravwojdyla**

### CODE REVIEW

**Comment by ravwojdyla** [comment]
Do not merge. Pending on #253 #259 

Todo:
 - [ ] add tests
 - [ ] try to expose CodecFactory instead of String for Codec (otherwise not easy to pass Codec Options)
 - [ ] contribute back to DataflowSDK?

**Comment by nevillelyh** [comment]
Is `CoderFactory` serializable? Otherwise it's hard to pass it. Don't wanna go full on Hadoop `Configuration` style for this. I'm happy with just `String codec` for now with some sane defaults.

File: build.sbt [hunk 1]
```diff
@@ -206,6 +207,7 @@ lazy val scioTest: Project = Project(
       "org.scalatest" %% "scalatest" % scalaTestVersion,
       // DataFlow testing requires junit and hamcrest
       "junit" % "junit" % junitVersion,
+      "com.novocode" % "junit-interface" % junitInterfaceVersion,
```
Comment (nevillelyh) [inline build.sbt]: What's this for?

Comment (ravwojdyla) [inline build.sbt]: Without it, I believe:
 * no JUnit tests are actually run via SBT
 * there is no feedback from JUnit tests

**Comment by nevillelyh** [comment]
LGTM. Should there be a non-null default though?

**Pull request merged**